### PR TITLE
Update last changes `yiisoft\validator`.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

- [x] Test error nested, each.
- [x] Improved `hasAttributes()`, and `getAtrributeNested()`.

### CODE REVIEW

File: src/FormModel.php [hunk 1]
```diff
@@ -196,18 +196,17 @@ public function setAttribute(string $name, $value): void
...
-        /** @var array<string, Resultset> $resultSet */
-        foreach ($resultSet as $attribute => $result) {
-            if ($result->isValid() === false) {
-                $this->formErrors->clear($attribute);
-                /** @psalm-suppress InvalidArgument */
-                $this->addErrors([$attribute => $result->getErrors()]);
-            }
+        $errors = $result->getErrorsIndexedByPath();
+
+        foreach ($errors as $attribute => $error) {
```
Comment (vjik) [inline src/FormModel.php:205]: Here can be problem in cases with nested data (if uses `Each`, `Nested` or custom rules that added errors with path).
`$errors` can be `{'attrName.0' => ['error1', 'error2']}`

Comment (terabytesoftw) [reply src/FormModel.php:205]: Something like this example:

Rules:
```php
final class FormWithNestedAttribute extends FormModel
{
    private ?int $id = null;
    private LoginForm $user;

    public function getRules(): array
    {
        return [
            'id' => [Required::rule()],
            'user.login' => [
                Required::rule(),
                HasLength::rule()->min(3)->tooShortMessage('Is too short.'),
            ],
        ];
    }
}
```

Test:
```php
$formModel = new FormWithNestedAttribute();
$validator = $this->createValidatorMock();
$this->assertTrue($formModel->load(['FormWithNestedAttribute' => ['user.login' => 'ad']]));
$this->assertFalse($validator->validate($formModel)->isValid());
$this->assertSame(
    ['Is too short.'],
    HtmlFormErrors::getErrorSummary($formModel, ['user.login']),
);
```

Comment (vjik) [reply src/FormModel.php:205]: I mean another case:

```php
public function getRules(): array
    {
        return [
            'id' => [Required::rule()],
            'names' => Each::rule(...),
        ];
    }
```

Comment (terabytesoftw) [reply src/FormModel.php:205]: Could you give me a real use case, of using an array in a form and validating each value.

Comment (vjik) [reply src/FormModel.php:205]: > Could you give me a real use case, of using an array in a form and validating each value.

Simple form with ID and array of labels.

File: src/FormModel.php [hunk 2]
```diff
@@ -251,6 +245,20 @@ protected function collectAttributes(): array
...
     }
 
+    /**
+     * @psalm-param non-empty-array<string, non-empty-list<int|string>> $items
+     */
+    private function addErrors(array $items): void
+    {
+        foreach ($items as $attribute => $errors) {
+            foreach ($errors as $error) {
+                if (is_string($error)) {
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/FormModel.php:255]: The error cannot be numeric or array at this level, what would be a real use case?

[the remark this answers is not in the document]
Comment (vjik) [reply src/FormModel.php:255]: > The error cannot be numeric or array at this level, what would be a real use case?

$errors passed to this method can be is non-string?

[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/FormModel.php:255]: https://github.com/yiisoft/validator/blob/99c2c7f22555e3e084cbaa78396cc0557f3cb24a/src/Result.php#L75
https://github.com/yiisoft/validator/blob/99c2c7f22555e3e084cbaa78396cc0557f3cb24a/src/Error.php#L25

Nope only string.

[the remark this answers is not in the document]
Comment (vjik) [reply src/FormModel.php:255]: Can delete condition `if (is_string($error)) {`.

[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/FormModel.php:255]: psaml it throws error for the annotation in yiisoft/validator the return is wrong.

[the remark this answers is not in the document]
Comment (vjik) [reply src/FormModel.php:255]: > psaml it throws error for the annotation in yiisoft/validator the return is wrong.
> 
> https://github.com/yiisoft/validator/blob/99c2c7f22555e3e084cbaa78396cc0557f3cb24a/src/Result.php#L68

This is should be fixed in validator.

[the remark this answers is not in the document]
Comment (arogachev) [reply src/FormModel.php:255]: https://github.com/yiisoft/validator/pull/168/commits/ec4d4a0fe708be7f668adc557525ce0fdd294c5a

:heavy_check_mark:

[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/FormModel.php]: fixed.

File: composer.json [hunk 3]
```diff
@@ -50,11 +50,7 @@
         }
     },
     "config": {
-        "sort-packages": true,
-        "allow-plugins": {
```
Comment (samdark) [inline composer.json:54]: Should not be removed.

**Review by samdark — APPROVED** [review]

File: composer.json (hunk 3 above)
Comment (terabytesoftw) [reply composer.json:54]: fixed.

**Pull request merged**